### PR TITLE
[#WA] [#DevOps] Microservice infrastructure

### PR DESCRIPTION
**Pull request by Aleksandra988**
- Web application refactored and split into 4 microservices.
- Docker files added/modified to support microservices
- Docker compose modified

### CODE REVIEW

File: MicroServiceAccount/Backend/Model/Admin.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            Password = password;
+            IsAdmin = isAdmin;
+        }
+        [JsonConstructor]
+        public Admin(string serialNumber, string name, string surname, string id, DateTime dateOfBirth, string contact, string email,
+            Address address, string password)
+            : base(serialNumber, name, surname, id, dateOfBirth, contact, email, address, password)
+        {
+
+        }
```
Comment (milicasiriski) [inline MicroServiceAccount/Backend/Model/Admin.cs]: Delete one of the JsonConstructor annotations, there can be only one or the app could break

File: MicroServiceAccount/Backend/Model/Patient.cs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        //public Patient(string serialNumber, string name, string surname, string id, DateTime dateOfBirth,
+        //    string contact, string email, Address address, string parentName, string gender, string password,
+        //    bool isGuest = false)
+        //    : base(serialNumber, name, surname, id, dateOfBirth, contact, email, address, password)
+        //{
+        //    ParentName = parentName;
+        //    Gender = gender;
+        //    Guest = isGuest;
+        //    Password = password;
+        //}
```
Comment (milicasiriski) [inline MicroServiceAccount/Backend/Model/Patient.cs]: Delete this if unused

File: MicroServiceAccount/Backend/Repository/DatabaseSql/MsAccountDbContext.cs [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+using MicroServiceAccount.Backend.Model.Util;
+using MicroServiceAccount.Backend.Repository.DatabaseSql.RelationHelpers;
+using Microsoft.EntityFrameworkCore;
+
+namespace MicroServiceAccount.Backend.Repository.DatabaseSql
+{
+    public class MsAccountDbContext : DbContext
+    {
+        //     private const string CONNECTION_STRING =
+        //          "User ID =postgres;Password=super;Server=localhost;Port=5432;Database=healthcare-system-db;Integrated Security=true;Pooling=true;";
```
Comment (milicasiriski) [inline MicroServiceAccount/Backend/Repository/DatabaseSql/MsAccountDbContext.cs]: You can delete this, it's not and should not be used

Comment (milicasiriski) [reply MicroServiceAccount/Backend/Repository/DatabaseSql/MsAccountDbContext.cs]: Okay now I see that the whole DbContext is moved and refactored here, I think the data should still be kept in the same database but extracted from different microservices (from their repositories?) but we can check with someone about that

File: MicroServiceAppointment/Backend/Model/Appointment.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        public Appointment(AppointmentDto appointmentDTO) : base()
+        {
+        //    Room = appointmentDTO.RoomDTO;
+        //    Physician = appointmentDTO.PhysicianDTO;
+        //    Patient = appointmentDTO.Patient;
+        //    TimeInterval = appointmentDTO.Time;
+        //    ProcedureType = appointmentDTO.ProcedureType;
+        //    Urgency = appointmentDTO.Urgency;
+        //    Date = appointmentDTO.Date;
+        }
```
Comment (milicasiriski) [inline MicroServiceAppointment/Backend/Model/Appointment.cs]: Delete this if unused

File: WebApplication/Controller/Enumerable.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+﻿namespace MicroServiceAccount.Backend.Controllers
+{
+    public class Enumerable<T>
+    {
+    }
```
Comment (milicasiriski) [inline WebApplication/Controller/Enumerable.cs]: Was this class created by accident?

**Review by milicasiriski — CHANGES_REQUESTED** [review]
As far as I can see it's mostly moving the files from your project into different microservices but we just need to check about the separate database contexts because of the docker deployment

File: WebApplication/Controller/Enumerable.cs (hunk 5 above)
Comment (Tanjaa1) [reply WebApplication/Controller/Enumerable.cs]: Yes :(

**Review by milicasiriski — COMMENTED** [review]
Looks ok now, I will change the docker files and it's good to go, and please try to make commit messages make more sense...

**Review by milicasiriski — APPROVED** [review]

**Review by lukaDoric — APPROVED** [review]

**Pull request merged**